src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -1459,30 +1489,30 @@ static s8 sLimbToBodyParts[DINOLFOS_LIMB_MAX] = {
 
 void EnDinofos_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx, Gfx** gfx) {
     EnDinofos* this = THIS;
-    Vec3f sp80;
-    Vec3f sp74;
-    Vec3f sp68;
-    Vec3f sp5C;
+    Vec3f vtx3;
+    Vec3f vtx2;
+    Vec3f vtx1;
+    Vec3f vtx0;
     s32 sp58;
-    Vec3f sp4C;
-    MtxF* sp48;
+    Vec3f fireVelocity;
+    MtxF* matrix;
 
     Collider_UpdateSpheres(limbIndex, &this->colliderJntSph);
     if (sLimbToBodyParts[limbIndex] != BODYPART_NONE) {
         Matrix_MultZero(&this->bodyPartsPos[sLimbToBodyParts[limbIndex]]);
     }
 
-    if ((limbIndex == DINOLFOS_LIMB_RIGHT_HAND) && (this->unk_292 != this->unk_290) &&
-        ((this->actionFunc == func_8089C56C) || (this->actionFunc == func_8089C2A8))) {
-        Math_Vec3f_Copy(&sp74, &this->colliderQuad.dim.quad[0]);
-        Math_Vec3f_Copy(&sp80, &this->colliderQuad.dim.quad[1]);
-        Matrix_MultVec3f(&D_8089E38C, &sp68);
-        Matrix_MultVec3f(&D_8089E398, &sp5C);
-        Collider_SetQuadVertices(&this->colliderQuad, &sp5C, &sp68, &sp74, &sp80);
-        if (this->colliderQuad.base.atFlags & AT_ON) {
-            EffectBlure_AddVertex(Effect_GetByIndex(this->unk_2A0), &sp68, &sp5C);
+    if ((limbIndex == DINOLFOS_LIMB_RIGHT_HAND) && (this->timer2 != this->timer1) &&
+        ((this->actionFunc == EnDinofos_SlashFromGround) || (this->actionFunc == EnDinofos_JumpSlash))) {
+        Math_Vec3f_Copy(&vtx2, &this->knifeCollider.dim.quad[0]);
+        Math_Vec3f_Copy(&vtx3, &this->knifeCollider.dim.quad[1]);
+        Matrix_MultVec3f(&D_8089E38C, &vtx1);
+        Matrix_MultVec3f(&D_8089E398, &vtx0);
+        Collider_SetQuadVertices(&this->knifeCollider, &vtx0, &vtx1, &vtx2, &vtx3);
```

Comment:
```suggestion
        Math_Vec3f_Copy(&knifeBaseQuadOldPos, &this->knifeCollider.dim.quad[0]);
        Math_Vec3f_Copy(&knifeTipQuadOldPos, &this->knifeCollider.dim.quad[1]);
        Matrix_MultVec3f(&sKnifeTipQuadOffset, &knifeTipQuadPos);
        Matrix_MultVec3f(&sKnifeBaseQuadOffset, &knifeBaseQuadPos);
        Collider_SetQuadVertices(&this->knifeCollider, &knifeBaseQuadPos, &knifeTipQuadPos, &knifeBaseQuadOldPos, &knifeTipQuadOldPos);
```
Some naming suggestions. Note I confirmed which quad vertex corresponded to the tip by using `func_800B10C0` to spawn dust at what you called `vtx1` and it spawned tip of the knife when they swung.